### PR TITLE
Main Menu + Main Scene basic implementation

### PR DESCRIPTION
**Pull request by havecookies**
Main Menu scene that moves you to the main scene when left click
Main scene that changes the room according to the .wmp file and changes to the next room on click

### CODE REVIEW

File: Core/WumpusCore/Controller/Controller.cs [hunk 1]
```diff
@@ -62,6 +86,26 @@ public string GetTriviaQuestion()
         {
             return minigameController.GetTriviaQuestion();
         }
-    }
 
+        public void StartGame()
+        {
+            ValidateScene(new []{StartScreen},InRoom);
```
Comment (ilhs1096069) [inline Core/WumpusCore/Controller/Controller.cs:97]: what is this

File: Core/WumpusCore/Controller/Controller.cs [hunk 2]
```diff
@@ -57,6 +91,26 @@ public Trivia.AskableQuestion GetTriviaQuestion()
...
+            this.state = InRoom;
+        }
+
+        /// <summary>
+        /// Throws an invalid operations exception if the current state is not in the valid states. Meant to be used as validation for methods to prevent UI from getting any funny ideas
```
Comment (ilhs1096069) [inline Core/WumpusCore/Controller/Controller.cs:102]: Please say what it's doing before saying the exception it'll throw if you do it wrong

File: Core/WumpusTesting/ControllerTest.cs
Comment (ilhs1096069) [inline Core/WumpusTesting/ControllerTest.cs:1]: wow exciting tests

File: Unity/Assets/Scripts/MainMenu UI.cs
Comment (ilhs1096069) [inline Unity/Assets/Scripts/MainMenu UI.cs:1]: Why is the file called "MainMenu _space_ UI" and not just MainMenuUI?

**Review by roguefirework — APPROVED** [review]
I see code that seems fine

File: Unity/Assets/Scripts/UI.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+using WumpusCore.Topology;
+
+public class UI : MonoBehaviour
+{
+    private Controller controller;
+
+    [SerializeField] 
+    private GameObject cam;
+
+    private const float camSens = 5f;
```
Comment (ilhs1096069) [inline Unity/Assets/Scripts/UI.cs:12]: This const is not following standard C# style standards, use PascalCase

Comment (ilhs1096069) [reply Unity/Assets/Scripts/UI.cs:12]: wow I am being mean today

**Review by ilhs1096069 — CHANGES_REQUESTED** [review]
There's a few things I mention in the review

File: Unity/Assets/Scripts/MainMenu UI.cs
Comment (havecookies) [reply Unity/Assets/Scripts/MainMenu UI.cs:1]: Because we can

File: Unity/Assets/Scripts/UI.cs (hunk 3 above)
Comment (havecookies) [reply Unity/Assets/Scripts/UI.cs]: okkkkkk......... fixed

**Review by MaximumMaxxx — APPROVED** [review]

**Review by ilhs1096069 — APPROVED** [review]
looks bad to me

**Pull request merged**